src/copy.js

```diff
@@ -7,20 +7,21 @@ chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
   }
 });
 
+/**
+ * Regex to match twitter and x.com URLs.
+ */
+const FIX_TARGET = /twitter\.com|x\.com/g
+
 document.addEventListener('copy', e => {
-  currentLink = document.location.href;
+  currentLink = window.location.href
+  target = window.getSelection().toString()
 
-  /* When the user visits the bookmarks page, and uses copy link button on a post, 
-    extension gets the url of the bookmarks page isntead of the post.
-    This is a temprary fix to get the url of the post instead.*/
+  if (!currentLink) {
+    return;
+  }
 
-  // TODO: Check more pages where this could be an issue.
-  if (currentLink.includes('bookmarks')) {
-    const postLink = e.target.innerHTML;
-    const urlRegex = /^https?:\/\/[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}(\/.*)?$/g;
-    if (postLink.match(urlRegex)) {
-      currentLink = postLink;
-    }
```

Comment:
This process is no longer necessary because the correction target is not retrieved from the link.